### PR TITLE
feat: replace jq with gojq

### PR DESCRIPTION
**Pull request by johejo**
Signed-off-by: Mitsuo Heijo <mitsuo.heijo@gmail.com>

related #7799

itchyny/gojq is well maintenanced and widely used. https://pkg.go.dev/github.com/itchyny/gojq?tab=importedby
argo-cd and argo-events already use gojq.

TestResourceTemplateWithOutputs seems fine.

Please do not open a pull request until you have checked ALL of these:

* [x] Create the PR as draft .
* [x] Run `make pre-commit -B` to fix codegen and lint problems.
* [x] Sign-off your commits (otherwise the DCO check will fail).
* [x] Use [a conventional commit message](https://www.conventionalcommits.org/en/v1.0.0/) (otherwise the commit message check will fail).
* [ ] "Fixes #" is in both the PR title (for release notes) and this description (to automatically link and close the issue).
* [x] Add unit or e2e tests. Say how you tested your changes. If you changed the UI, attach screenshots.
* [x] Github checks are green.
* [x] Once required tests have passed, mark your PR "Ready for review".

If changes were requested, and you've made them, dismiss the review to get it reviewed again.

### CODE REVIEW

File: workflow/executor/resource.go [hunk 1]
```diff
@@ -311,14 +311,37 @@ func (we *WorkflowExecutor) SaveResourceParameters(ctx context.Context, resource
 		}
 		output := string(out)
 		if param.ValueFrom.JQFilter != "" {
```
Comment (NikeNano) [inline workflow/executor/resource.go:313]: This looks really nice! Are the some tests to verify this logic? Maybe we could add one if not some old?

**Review by NikeNano — APPROVED** [review]
LGTM

**Review by JPZ13 — APPROVED** [review]
Looks great! @rohankmr414 - could we get your eyes on this PR too? I know you were running into jq issues previously

**Review by terrytangyuan — COMMENTED** [review]
Could you also add an integration/e2e test?

**Comment by johejo** [comment]
The e2e test case using jq Filter already exists, should I add some tests case that has almost same meanings? 
#10150 
https://github.com/argoproj/argo-workflows/blob/897b06a2c336990792606fca379ab4ef4fa58da3/test/e2e/resource_template_test.go#L146

Or could you suggest me other test perspectives that I have not realized.
Thanks.

**Review by rohankmr414 — APPROVED** [review]
LGTM! Thank you for doing this @johejo ❤️

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**